src/main/java/ch/njol/skript/lang/SkriptParser.java

```diff
@@ -577,6 +580,37 @@ public boolean hasTag(String tag) {
 				log.printError();
 				return null;
 			}
+			Matcher literalClassInfoMatcher = LITERAL_CLASSINFO_PARSE.matcher(expr);
+			if (literalClassInfoMatcher.matches()) {
+				String literalString = literalClassInfoMatcher.group(1);
+				String classInfoString = literalClassInfoMatcher.group(2);
+				ClassInfo<?> classInfo = Classes.parse(classInfoString, ClassInfo.class, context);
+				if (classInfo == null) {
+					log.printError();
+					return null;
+				}
+				boolean canParse = false;
+				for (ClassInfo<?> exprClassInfo : exprInfo.classes) {
+					if (exprClassInfo.getC().isAssignableFrom(classInfo.getC())) {
+						canParse = true;
+						break;
+					}
+				}
+				if (!canParse) {
+					log.printError();
+					return null;
+				}
```

Comment:
I'd like to maybe see an error here that says something like "This must be a <accepted classinfos>, not a <given classinfo>" or whatever the precedent is for those error messages.